lib/publiccloud/azure.pm

```diff
@@ -124,16 +131,61 @@ sub upload_img {
 
     my $key = $self->get_storage_account_keys($storage_account);
 
-    assert_script_run('az storage blob upload --max-connections 4 --account-name '
-          . $storage_account . ' --account-key ' . $key . ' --container-name ' . $self->container
-          . ' --type page --file ' . $file . ' --name ' . $img_name, timeout => 60 * 60 * 2);
-    assert_script_run('az disk create --resource-group ' . $self->resource_group . ' --name ' . $disk_name
-          . ' --source https://' . $storage_account . '.blob.core.windows.net/' . $self->container . '/' . $img_name
-          . ' --hyper-v-generation=' . $sku . ' --architecture=' . $arch);
 
-    assert_script_run('az image create --resource-group ' . $self->resource_group . ' --name ' . $img_name
-          . ' --os-type Linux --hyper-v-generation=' . $sku . ' --source=' . $disk_name);
 
+    # Check if blob already exists
+    my $container = $self->container;
+    my $blobs = script_output("az storage blob list --container-name '$container' --account-name '$storage_account' | jq '.[].name'");
+    $blobs =~ s/^\s+|\s+$//g;    # trim
+    record_info("blobs", $blobs);
+    my @blobs = split(/\n/, $blobs);
+    if (grep(/$img_name/, @blobs)) {
+        record_info('blob', "Blob already exists, omitting upload");
+    } else {
+        # Upload blob to Azure. Keep in mind that VM images need to be a page blob type
+        assert_script_run('az storage blob upload --max-connections 4 --account-name '
+              . $storage_account . ' --account-key ' . $key . ' --container-name ' . $self->container
+              . ' --type page --file ' . $file . ' --name ' . $img_name, timeout => 60 * 60 * 2);
+    }
+
+    if ($arch eq 'Arm64') {
+        # For Arm64 images we need to use the image galleries
+        my $publisher = get_var("PUBLIC_CLOUD_AZURE_PUBLISHER", "qe-c");
+        my $offer = get_required_var("PUBLIC_CLOUD_AZURE_OFFER");
+        my $definition = get_required_var('DISTRI') . '-' . get_required_var('FLAVOR') . '-' . get_required_var('VERSION');
+        $definition = get_var("PUBLIC_CLOUD_AZURE_IMAGE_DEFINITION", uc($definition));
+        $sku = get_required_var("PUBLIC_CLOUD_AZURE_SKU");
+        my $target_regions = get_var("PUBLIC_CLOUD_REGION", "westeurope");
+        $target_regions =~ s/,/ /;    # CLI expects spaces as separation, not commas
+        my $hyperv = $sku =~ 'gen1' ? 'V1' : 'V2';
```

Comment:
@pdostal  I remember you were doing something in same direction recently but can not catch up what exactly ...
do you see any code duplication here ?

Reply:
Please see also line 157, where we need the `sku` variable. If at all, the usage of `sku` in line 121 is wrong, because it is not the sku but the hyper-v version based on the sku.

Reply:
I agree with Felix The condition on line 121 is not 100% right and $sku = get_required_var("PUBLIC_CLOUD_AZURE_SKU"); looks like better approach to me.

Reply:
I would be nice if you bring order in this variable naming including line 121 . also it would be better from readability PoV all activity around $sku/$hyperv will happen in one place . But feel free to ignore this request ...